app/src/main/cpp/DeviceUtils.cpp

```diff
@@ -60,6 +63,44 @@ void DeviceUtils::GetTargetImmersiveSize(const uint32_t aRequestedWidth,
   aTargetHeight = (uint32_t) fmaxf(fminf(aRequestedHeight, aMaxHeight), minHeight);
 }
 
+vrb::GeometryPtr DeviceUtils::GetSphereGeometry(vrb::CreationContextPtr& context, uint32_t resolution, float radius)
+{
+    vrb::VertexArrayPtr array = vrb::VertexArray::Create(context);
+    vrb::GeometryPtr geometry = vrb::Geometry::Create(context);
+    vrb::TransformPtr transform = vrb::Transform::Create(context);
+    std::vector<int> indices;
+
+    int rings = (int) resolution;
+    int sectors = (int) resolution;
+    float const R = 1.0f / ((float) rings - 1.0f);
+    float const S = 1.0f / ((float) sectors - 1.0f);
+
+    for (int r = 0; r < rings; r++) {
+        for (int s = 0; s < sectors; s++) {
+            float const y = sinf(- (float) M_PI_2 + (float) M_PI * (float) r * R);
+            float const x = cosf(2.0f * (float) M_PI * (float) s * S) * sinf(M_PI * (float) r * R);
+            float const z = sinf(2.0f * (float) M_PI * (float) s * S) * sinf(M_PI * (float) r * R);
+            array->AppendVertex(vrb::Vector(x * radius, y * radius, z * radius));
+            array->AppendNormal(vrb::Vector(x, y, z).Normalize());
+            array->AppendUV(vrb::Vector((float)s * S, (float)r * R, 0.0));
+        }
+    }
+
+    geometry->SetVertexArray(array);
+
+    for (int r = 0; r < rings; r++) {
```

Comment:
Can't we merge these two loops? Or is it because `setVertexArray` must be called before `AddFace`?

Reply:
Nope, unfortunately we can't merge them. `AddFace` needs the vertex array already set.